test/SparqlAntlrParserTest.cpp

```diff
@@ -136,3 +137,69 @@ TEST(SparqlExpressionParser, First) {
   AD_CHECK(std::holds_alternative<double>(result));
   ASSERT_FLOAT_EQ(25.0, std::get<double>(result));
 }
+
+TEST(SparqlParser, ComplexConstructQuery) {
+  string input =
+      "CONSTRUCT { [?a ( ?b (?c) )] ?d [?e [?f ?g]] . "
+      "<http://wallscope.co.uk/resource/olympics/medal/#something> a "
+      "<http://wallscope.co.uk/resource/olympics/medal/#somethingelse> } "
+      "WHERE {}";
+  ParserAndVisitor p{input};
+
+  auto triples = p.parser.constructQuery()
+                     ->accept(&p.visitor)
+                     .as<std::vector<std::array<VarOrTerm, 3>>>();
+  ASSERT_EQ(triples.size(), 11u);
+  auto nil = "<http://www.w3.org/1999/02/22-rdf-syntax-ns#nil>";
+  auto first = "<http://www.w3.org/1999/02/22-rdf-syntax-ns#first>";
+  auto rest = "<http://www.w3.org/1999/02/22-rdf-syntax-ns#rest>";
+
+  EXPECT_THAT(triples[0][0], IsBlankNode("_:g_0"));
+  EXPECT_THAT(triples[0][1], IsVariable("?a"));
+  EXPECT_THAT(triples[0][2], IsBlankNode("_:g_3"));
```

Comment:
`EXPECT_THAT(triples[0], {Blank(true, "0"), Var("?a"), Blank(true, "3")});`
or even
`expectTriple(0, {Blank(true, "0"), Var("?a"), Blank(true, "3")})`;

Reply:
and all Blank nodes are generated, so this can be also abstracted away

Reply:
`blank(0)`